README.md

```diff
@@ -2,27 +2,90 @@
+- [Reference](docs/REFERENCE.md)
+- [Usage](docs/USAGE.md)
+- [Contributing](docs/CONTRIBUTING.md)
 
-## Supported versions
+# Introduction
+
+This package aims to make it easier for developers using StreamField who need to write data
+migrations when making changes involving blocks/block structure in the StreamField. We expose a
+custom migration operation class (`MigrateStreamData`) for migrations, which recurses through
+a streamfield to apply chosen sub-operations to all blocks matching a specific type. With it we also 
+supply a set of sub-operations to perform the most common changes, while also allowing you to
+write your own when needed.
+
+## Why Data Migrations?
+
+A `StreamField` is stored as a single column of JSON data in the database. Blocks are stored as structures within the JSON, and can be nested. However, as far as django is concerned when making schema migrations, everything inside this column is just a string of JSON data and the schema doesn’t change regardless of the content/structure of the StreamField since it is the same field type before and after the change. Therefore whenever changes are made to StreamFields, any existing data must be changed into the required structure by using a data migration created manually by the user. If
+the data is not migrated, even a simple change like renaming a block can result in old data being lost.
+
+Generally, data migrations are done manually by making an empty migration file and writing the forward and backward functions for a RunPython command which will handle the logic for taking the previously saved JSON representation and converting it into the new JSON representation needed.
 
-- Python ...
-- Django ...
-- Wagtail ...
+While this is fairly straightforward for a very simple and small change like changing the name of a CharBlock, where you could define two functions for mapping the data and loop through all the blocks in the StreamField checking whether they are the required block type and applying the mapper accordingly; this can easily get very complicated when nested blocks and multiple fields are involved.
+
+This package would make it easier to write data migrations for streamfield changes by providing utilities
+to recurse through various streamdata structures and map changes, in addition to having several "operations"
+which also handle the logic for altering the data for common use cases like renaming, removing and altering
+values of blocks.
+
+# Quick Start
 
 ## Installation
 
-- `pip install streamfield-migration-toolkit`
-- ...
+- `pip install wagtail-streamfield-migration-toolkit`
+- Add `"wagtail_streamfield_migration_toolkit"` to `INSTALLED_APPS`
+
+## Supported versions
+
+- Python 3.7, 3.8, 3.9
+- Django 3.2, 4.0
+- Wagtail 4.0
+
+## Quick Usage
+
+Assume we have a model `BlogPage` in app `blog` which has a streamfield `content` with a child
+streamblock named `mystream` which has a child char block named `field1` which we want to rename to
```

Comment:
This is a little hard to follow. Perhaps we could just write the code?

```suggestion
Assume we have a model `BlogPage` in app `blog`, defined as follows:

```python
class BlogPage(Page):
    content = StreamField([("field1", blocks.CharBlock())])```

If we want to rename "field1" to "block1", we would use the following migration
```